tsconfig.json

```diff
@@ -1,10 +1,36 @@
 {
   "compilerOptions": {
-    // Not sure why, but this is needed for typeRoots to work correctly
-    // Source: https://github.com/microsoft/TypeScript/issues/22217#issuecomment-369783776
+    "allowJs": false,
+    "allowSyntheticDefaultImports": false,
+    "checkJs": false,
+    "declaration": true,
+    "declarationMap": true,
+    "emitDecoratorMetadata": true,
+    "esModuleInterop": true,
+    "experimentalDecorators": true,
+    "forceConsistentCasingInFileNames": true,
+    "incremental": true,
+    "isolatedModules": true,
+    "jsx": "react",
+    "lib": ["es2017", "DOM"],
+    "module": "commonjs",
+    "moduleResolution": "node",
+    "newLine": "lf",
+    "noEmitOnError": false,
+    "noImplicitAny": true,
+    "noResolve": false,
+    "noUnusedLocals": true,
+    "noUnusedParameters": true,
     "paths": {
       "*": ["./@types/*"]
     },
-    "typeRoots": ["./@types", "node_modules/@types"]
-  }
+    "resolveJsonModule": true,
+    "skipLibCheck": true,
+    "sourceMap": true,
+    "suppressImplicitAnyIndexErrors": true,
+    "target": "es6",
+    "typeRoots": ["./@types", "node_modules/@types"],
+    "strict": true,
+  },
+  "exclude": ["node_modules", "docs", "dist", "__testfixtures__"]
```

Comment:
So this is needed by storybook when compiling typescript and type checking. It's a copy of the component based tsconfigs, but with some of the extra settings that were missed from our previous set up

Reply:
Can you move these to the storybook folder? the package tsconfigs extend this one, so I think it would be best to keep this root level tsconfig as generic as possible.

Reply:
I can, but my only worry is that the config in the packages now misses the following, not set by default, settings that we used to have for all packages:

```
"esModuleInterop": true,
"forceConsistentCasingInFileNames": true,
"newLine": "lf",
"noUnusedLocals": true,
"noUnusedParameters": true,
"resolveJsonModule": true,
"strict": true,
```

Some of which are pretty important and actually caught some things, like unused type imports for combobox that eslint isn't picking up.